components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsSearchBox/src/EIAnalyticsSearchBox.jsx

```diff
@@ -176,4 +156,20 @@ class EIAnalyticsSearchBox extends Widget {
     }
 }
 
+function getCurrentPage() {
```

Comment:
`getCurrentPage`, `getKey` functions are repeated several times. Lets move them into a single place (a `class`) and `import` where necessary and use.